app/src/main/java/com/bignerdranch/android/androidprojecttms/Delivery.kt

```diff
@@ -0,0 +1,46 @@
+package com.bignerdranch.android.androidprojecttms
+
+sealed class Delivery(var deliver: (Order) -> Unit) {
+    data class DpdBy(val coast: Int) : Delivery({
+        println(
+            """
+            Deliver: Dpd.by
+            Id: ${it.getIdOrder()}
+            Price: ${it.getPrice()}
+            
+            """.trimIndent()
+        )
+    })
+
+    data class BoxAgentCom(val coast: Int) : Delivery({
+        println(
+            ("""
+            Deliver: BoxAgent.com
+            Id: ${it.getIdOrder()}
+            Product: ${it.getProduct()}
+            Price: ${it.getPrice()}
+            
+            """).trimIndent()
+        )
+    })
+
+    data class VozimBy(val coast: Int) : Delivery({
```

Comment:
If all your `Delivery` child classes have the same field, it's reasonable to add it to the parent (and write inside println function call).
It will be `open val coast: Int` for `Delivery` and `override val coast: Int` for all three children.